iceoryx_platform/mac/include/iceoryx_platform/attributes.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_MAC_PLATFORM_ATTRIBUTES_HPP
+#define IOX_HOOFS_MAC_PLATFORM_ATTRIBUTES_HPP
+
+/// @brief IOX_NO_DISCARD adds the [[nodiscard]] keyword if it is available for the current compiler.
+///        If additionally the keyword [[gnu::warn_unused]] is present it will be added as well.
+/// @note
+//    [[nodiscard]], [[gnu::warn_unused]] supported since gcc 4.8 (https://gcc.gnu.org/projects/cxx-status.html)
+///   [[nodiscard]], [[gnu::warn_unused]] supported since clang 3.9 (https://clang.llvm.org/cxx_status.html)
+///   activate keywords for gcc>=5 or clang>=4
+
```

Comment:
On mac os we support only clang and always compile with C++17. Therefore you can use the following:
```cxx
#define IOX_NO_DISCARD [[nodiscard, gnu::warn_unused]]
#define IOX_FALLTHROUGH [[fallthrough]]
```

and remove all the other macro ifs.

Reply:
@elfenpiff Good point. Done :heavy_check_mark: